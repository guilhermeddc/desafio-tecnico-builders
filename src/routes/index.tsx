import React from 'react';
import {BrowserRouter} from 'react-router-dom';

import {PrivateRoutes} from './private.routes';

export const Routes: React.FC = () => {
  return (
    <BrowserRouter>
      <PrivateRoutes />
    </BrowserRouter>
  );
};
