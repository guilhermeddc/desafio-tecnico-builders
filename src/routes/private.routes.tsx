import React, {lazy, Suspense} from 'react';
import {Routes, Route, Navigate} from 'react-router-dom';

import {LinearDeterminate} from 'shared/components';
import {BaseLayout} from 'shared/styles/layouts';

const Home = lazy(() => import('pages/Home'));

export const PrivateRoutes: React.FC = () => {
  return (
    <BaseLayout>
      <Suspense fallback={<LinearDeterminate />}>
        <Routes>
          <Route path="/" element={<Home />} />

          <Route path="*" element={() => <Navigate to="/" replace />} />
        </Routes>
      </Suspense>
    </BaseLayout>
  );
};
