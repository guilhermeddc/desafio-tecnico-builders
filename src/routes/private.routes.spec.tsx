import {MemoryRouter} from 'react-router-dom';

import {render} from '@testing-library/react';

import {PrivateRoutes} from './private.routes';

describe('PrivateRoutes', () => {
  it('should render the PrivateRoutes component and redirect to home', () => {
    const {container} = render(
      <MemoryRouter initialEntries={['/test']} initialIndex={0}>
        <PrivateRoutes />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
