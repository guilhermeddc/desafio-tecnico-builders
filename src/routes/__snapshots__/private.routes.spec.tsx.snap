// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoutes should render the PrivateRoutes component and redirect to home 1`] = `
<div>
  <main
    class="MuiBox-root css-dgaybt"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
    >
      <div
        class="MuiBox-root css-15gobu2"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter css-149xwtl-MuiTypography-root"
        >
          ☀️ Desafio Técnico
        </h4>
        <span
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="0"
          class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-10zxdc7-MuiLinearProgress-root"
          role="progressbar"
        >
          <span
            class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-5xe99f-MuiLinearProgress-bar1"
            style="transform: translateX(-100%);"
          />
        </span>
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter css-1kgxqm0-MuiTypography-root"
        >
          Builders
        </h6>
      </div>
    </div>
  </main>
</div>
`;
